src/puppetlabs/puppetdb/scf/hash.clj

```diff
@@ -95,11 +96,19 @@
 (s/defn package-similarity-hash :- s/Str
   "Creates a stable ordering of `packages` and computes a hash over
   that structure"
-  [packages :- [package-tuple]]
-  (-> packages
-      sort
-      json/generate-string
-      kitchensink/utf8-string->sha1))
+  [packages :- [hashed-package-tuple]]
+  (->> packages
+       (map package-tuple-hash)
+       sort
```

Comment:
That uses `ks/sort-nested-maps` for sorting, which doesn't sort sequential collections. We need to sort the sequence here.